### PR TITLE
Skip BackupSession creation if target does not exist + use timestamp …

### PR DESCRIPTION
**Pull request by hossainemruz**
Fixes: https://github.com/stashed/stash/issues/767, https://github.com/stashed/stash/issues/791

### CODE REVIEW

File: hack/deploy/rbac-list.yaml [hunk 1]
```diff
@@ -39,6 +39,12 @@ rules:
   - replicasets
   - statefulsets
   verbs: ["get", "list", "watch", "patch"]
+- apiGroups:
```
Comment (tamalsaha) [inline hack/deploy/rbac-list.yaml]: Why do we need this?

File: pkg/controller/rbac_jobs.go [hunk 2]
```diff
@@ -74,12 +76,42 @@ func (c *StashController) ensureCronJobClusterRole(psps []string) error {
...
 				Verbs:         []string{"use"},
 				ResourceNames: psps,
 			},
+			{
+				APIGroups: []string{apps.GroupName},
+				Resources: []string{"deployments", "statefulsets", "replicasets", "daemonsets"},
+				Verbs:     []string{"get"},
+			},
+			{
+				APIGroups: []string{extensions.GroupName},
```
Comment (tamalsaha) [inline pkg/controller/rbac_jobs.go]: Why do we need both?

File: pkg/util/util.go [hunk 3]
```diff
@@ -281,3 +287,101 @@ func ParseInt32P(v string) (*int32, error) {
...
+}
+
+func (wc *WorkloadClients) IsTargetExist(target api_v1beta1.TargetRef, namespace string) bool {
+
+	fmt.Println(target.APIVersion, target.Kind, target.Name, namespace)
+	switch target.Kind {
+	case apis.KindDeployment:
+		switch target.APIVersion {
+		case apis.APIVersionAppsV1:
+			if _, err := wc.KubeClient.AppsV1().Deployments(namespace).Get(target.Name, metav1.GetOptions{}); err == nil {
```
Comment (tamalsaha) [inline pkg/util/util.go]: You don't need to check both. Just use apps/v1 regardless of what user says for Official workloads.

File: hack/deploy/rbac-list.yaml (hunk 1 above)
Comment (hossainemruz) [reply hack/deploy/rbac-list.yaml]: removed

File: pkg/controller/rbac_jobs.go (hunk 2 above)
Comment (hossainemruz) [reply pkg/controller/rbac_jobs.go]: removed

File: pkg/util/util.go (hunk 3 above)
Comment (hossainemruz) [reply pkg/util/util.go]: removed

**Pull request merged**